dialogue-target/src/main/java/com/palantir/dialogue/Clients.java

```diff
@@ -0,0 +1,39 @@
+/*
+ * (c) Copyright 2020 Palantir Technologies Inc. All rights reserved.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package com.palantir.dialogue;
+
+import com.google.common.util.concurrent.ListenableFuture;
+
+/**
+ * Provides functionality for generated code to make both blocking and asynchronous calls without
+ * duplicating logic.
+ */
+public interface Clients {
+
+    /**
+     * Makes a request to the specified {@link Endpoint} and deserializes the response using a provided deserializer.
+     */
+    <T> ListenableFuture<T> call(Channel channel, Endpoint endpoint, Request request, Deserializer<T> deserializer);
+
+    /**
+     * Semantics match {@link #call(Channel, Endpoint, Request, Deserializer)} but also blocks until
+     * the {@link ListenableFuture} has completed for blocking clients.
+     * This approach is used instead of using a separate method to block on a single future to simplify
+     * otherwise redundant generated code.
+     */
+    <T> T blocking(Channel channel, Endpoint endpoint, Request request, Deserializer<T> deserializer);
```

Comment:
this interface doesn't quite work with the existing codegen. It would force us to duplicate the code for building the request in the blocking/async impls

Reply:
Ah doh, our sample services are no longer representative

Reply:
I could go either way naming `<T> T blocking(ListenableFuture<T>)`. Preference on `return clients().block(future)` vs `return clients().blocking(future);

Reply:
I think i prefer `block` so that both methods are verbs

Reply:
:+1: thanks!